packages/compiler/core/scanner.ts

```diff
@@ -167,7 +167,6 @@ export const enum KeywordLimit {
   // have to change the keyword lookup algorithm in that case.
   MaxLength = 9,
```

Comment:
Is this wrong then? or it is another problem that also happens after 10 char

Reply:
10 is the real limit after this change. After that, we will hit the safe integer max. We hit 6 because of a different JavaScript trap where integers have max safe 32-bit range, not ~53-bit when you use bitwise operations.

Reply:
It's sad that I can't do efficient 64-bit integer math in 2022 in JS.